### ISSUE TITLE
Reset history for version 2

### ISSUE THREAD

**Issue by infradig** [trealla-prolog/trealla#1]
Time for a clean slate after two years of code-cruft. A gigabyte of git commit history was just ridiculous.

**Comment by infradig** [comment]
I will restore the old repo though in a while, for reference purposes.

**Comment by pabs3** [comment]
Hmm, I cloned [a fork of the old repo](https://github.com/guregu/trealla) and got 8MB and 7448 commits. Where does the 1GB come from?

**Comment by infradig** [comment]
Probably because it's only main. 99% of work happened in devel or other
specialist branches.

**Comment by infradig** [comment]
![Screenshot from 2022-08-09 11-54-09](https://user-images.githubusercontent.com/22718738/183546764-340d4e31-61c8-4e7b-890a-48cf6412de23.png)

**Comment by infradig** [comment]
Maybe some junk got in there somehow, somewhere.

**Comment by infradig** [comment]
It seems all the space is taken by many, many .pack files, the largest being 400MB.

**Comment by pmoura** [comment]
`git gc`?

**Comment by infradig** [comment]
Maybe would have worked, saw a lot of conflicting advice on that. Anyway I
like a fresh bowl start, feels like a load off my back.

**Comment by flexoron** [comment]
Perhaps contains more than 3 million json files 
$ git rev-list --objects --all | grep .json | wc -l
3285524

For example: data/test-cases/e60dce80b7e4576f.json

Maybe avoidable

Decreased now (v2.0.1)
$ git rev-list --objects --all | grep .json | grep test-cases | wc -l
16798

**Comment by infradig** [comment]
And I have no idea what they are.

**Comment by flexoron** [comment]
There is a `trealla/.github/workflows/compliance.yml` file which tries to generate an Allure-Report(whatever that is)
Click `Actions` above (Code Issues Pull requests `Actions` Security Insights)
```
All workflows

Pages build and deployment by BOT 6 hours ago

Fix setof/3 to fail (like bagof/3) ... testing 13 hours ago

Initial commit ... testing yesterday

```
As you can see a `BOT` processes commits and fills up the gh-pages branch 
But don't ask me why it has been configured.
Perhaps you can disable it in Settings/Pages (Repo configurations area)

**Comment by flexoron** [comment]
This workflows/`compliance.yml` file is an ansible 'script'
and every time you push something into the 'testing' branch,later on a build process(an action) starts (somewhere on github I guess), but do you really need these runs?
I assume this 'allure-report' generation creates all these files in the gh-pages directory.

https://github.com/peaceiris/actions-gh-pages
Set a branch name to use as GitHub Pages branch. The default is gh-pages.

Also see/ask @pmoura: logtalk_allure_report

**Comment by flexoron** [comment]
Here are the json files:
```
Actions -> Fix setof/3 to fail (like bagof/3) -> compliance.yml on push -> test ->
Upload Allure Report -> Create a commit:

  /usr/bin/git commit -m deploy: 035aea994d7b215010b6c6d820d9b05382369b53
  Auto packing the repository in background for optimum performance.
  See "git help gc" for manual housekeeping.
  [gh-pages 11a0fef] deploy: 035aea994d7b215010b6c6d820d9b05382369b53
   16820 files changed, 721462 insertions(+), 721451 deletions(-)
  : 
  : 
```
Perhaps logtalk test results?

**Comment by pmoura** [comment]
In the old repo, Logtalk distribution tests (which include the Prolog standard compliance suite) were run when changes were pushed to the `testing` branch. The resulting report was published using GitHub Pages. Regularly pruning the history of the `gh-pages` would have avoided the perceived bloat in the `gh-pages` branch.

**Comment by UWN** [comment]
> I will restore the old repo though in a while, for reference purposes.

This will not restore the cross-links with other projects AFAIK. They are just lost. So for the future, it does not make sense to contribute on your github pages.

**Comment by infradig** [comment]
These cross links are ephemeral anyway. I have for a while contemplated a
move from Github, it's not what it used to be. But not yet.

**Comment by flexoron** [comment]
https://sfconservancy.org/GiveUpGitHub/

**Comment by infradig** [comment]
Yes, I have never had a problem with hosting on a proprietary platform, but
being owned and used by Microsoft I do find a problem. Gitlab is
stand-alone and has certain advantages (such as familiarity). Other options
might be worth looking into.

**Comment by flexoron** [comment]
Berlin,Berlin https://codeberg.org/
Amsterdam  https://sourcehut.org/

btw 
Look at this: a sourcehut development.
Really cool video, 4 minutes
https://aerc-mail.org/

**Comment by UWN** [comment]
In any case, all the issues are lost.

**Comment by UWN** [comment]
It [seems](https://stackoverflow.com/questions/9247961/recover-a-deleted-repository-github-issues) there are ways to recover issues.

**Comment by UWN** [comment]
@infragdig: According to a response from github, only the owner can restore a repository including their issues. And that within 90 days. So you would need to restore it somehow.

**Comment by infradig** [comment]
I've got a spare PI/4, and I could use that with Gitea 😎
